include/swift/AST/Module.h

```diff
@@ -225,6 +225,13 @@ class ModuleDecl : public TypeDecl, public DeclContext {
     return { Files.begin(), Files.size() };
   }
 
+  void forEachSourceFile(const std::function<void(SourceFile &)> &fn) {
```

Comment:
Hmm. I'm not sure I like this because most modules don't have any source files, and it doesn't make sense to call this on those. It's only interesting on the main module today.

\*continues sweeping `-enable-source-import` under the rug\*

On a more mundane note, we have a better type for functions that don't get persisted: `llvm:function_ref`. It's the equivalent of ArrayRef for callbacks.

Reply:
Fixed the function type to be function_ref, tx.

Wrt not liking forEachSourceFile, I'm confused by your uncertainty. The function does do exactly what it says, and I pulled this out in response to an earlier comment of yours: 

"jrose-apple 6 days ago Owner
Last time there was a loop like this you decided to factor it out. Maybe there should be a higher-order forEachFileToEmitCodeFor or something?"

If you can just tell me what you would like, I'll be happy to do it.

Reply:
The function is a reasonable thing to have, just not on Module. It is highly likely that the only clients would be in CompilerInvocation, and it'd be a trap for people who don't realize that they might need to consider non-SourceFiles for their particular use.

Reply:
I've moved the function into frontEnd.cpp as a file static, and added "In" to its name, since the first argument is now the module.